File: integrations/java/oicana/src/main/java/com/oicana/manifest/PackageInfo.java
package com.oicana.manifest;

import java.util.List;

/**
 * The Typst package a template is.
 *
 * @param name Name of the template
 * @param version Version of the template
 * @param entrypoint File the compilation starts at
 * @param authors Authors of the template
 * @param license License of the template, or {@code null}
 * @param description Short description of the template, or {@code null}
 * @param homepage Web presence of the template, or {@code null}
 * @param repository Repository the template is developed in, or {@code null}
 */
public record PackageInfo(
        String name,
        String version,
        String entrypoint,
        List<String> authors,
        String license,
        String description,
        String homepage,
        String repository) {}

File: integrations/php/oicana-php/src/Manifest/PackageInfo.php
<?php

declare(strict_types=1);

namespace Oicana\Manifest;

/**
 * The Typst package a template is.
 */
final readonly class PackageInfo
{
    /**
     * @param string $name Name of the template
     * @param string $version Version of the template
     * @param string $entrypoint File the compilation starts at
     * @param list<string> $authors Authors of the template
     * @param string|null $license License of the template
     * @param string|null $description Short description of the template
     * @param string|null $homepage Web presence of the template
     * @param string|null $repository Repository the template is developed in
     */
    public function __construct(
        public string $name,
        public string $version,
        public string $entrypoint,
        public array $authors,
        public ?string $license,
        public ?string $description,
        public ?string $homepage,
        public ?string $repository
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            $data['name'],
            $data['version'],
            $data['entrypoint'],
            $data['authors'],
            $data['license'],
            $data['description'],
            $data['homepage'],
            $data['repository']
        );
    }
}
